setup.py

```diff
@@ -5,38 +5,53 @@
     find_packages,
 )
 
-extras_require = {
-    'test': [
-        "pytest==3.3.2",
+
+def merge_lists(base_list, *other_lists):
+    new_list = base_list[:]
+
+    for other_list in other_lists:
+        new_list.extend(i for i in other_list if i not in new_list)
+
+    return new_list
+
+
+HYPOTHESIS_REQUIREMENT = "hypothesis>=3.6.1"
+
+extras_require = {}
+extras_require['tools'] = [
+    HYPOTHESIS_REQUIREMENT,
+]
+extras_require['test'] = merge_lists(
+    [
+        "pytest==4.4.1",
         "pytest-pythonpath>=0.7.1",
         "pytest-xdist==1.22.3",
-        "hypothesis>=3.6.1",
         "tox>=2.9.1,<3",
         "eth-hash[pycryptodome]",
+        HYPOTHESIS_REQUIREMENT,
     ],
-    'lint': [
-        "flake8==3.4.1",
-        "isort>=4.2.15,<5",
-        "mypy==0.620",
-    ],
-    'doc': [
-        "Sphinx>=1.6.5,<2",
-        "sphinx_rtd_theme>=0.1.9",
-    ],
-    'dev': [
+    extras_require['tools'],
```

Comment:
I'm not sure `tools` needs to be merged into `test`. We can add `tools` as an explicit extra in the jobs that need it, and hypothesis is already explicitly added.

Reply:
Would that make it harder for contributors to run tests locally without using tox i.e. by just doing `pytest`?  The job extras only get added by tox.  If the `tools` submodule adds requirements other than hypothesis, then you wouldn't be able to just do a vanilla `pytest` run without getting import errors.

Reply:
I guess I have an expectation that, if I do `pip install -e .[test]`, then I've got everything I need to run the tests with `pytest`.  I also expect that the test suite covers everything, including optional submodules.

Reply:
> Would that make it harder for contributors to run tests locally without using tox i.e. by just doing `pytest`? The job extras only get added by tox. If the `tools` submodule adds requirements other than hypothesis, then you wouldn't be able to just do a vanilla `pytest` run without getting import errors.

I think this scenario is what `[dev]` is for. Sometimes extras are even mutually exclusive, like `eth-hash[pycryptodome]` and `eth-hash[pysha3]`, so there's no way to totally abstract away the things that tox is doing. My approach is that `[test]` is for adding the tools you need to run tests, not the things *being tested*. So if you want to test some extra that is not installed by default, then tox (or you, if outside of tox) will need to know to install it explicitly, or let `[dev]` install it for you.

In the end, this is all very academic, since `hypothesis` is installed by tests anyway. So I think we can kick the can down the road until tools has something that's not in `tests`.

Reply:
> My approach is that [test] is for adding the tools you need to run tests, not the things being tested.

Gotcha, makes sense.

> So I think we can kick the can down the road until tools has something that's not in tests.

👍